File: leetcode/ClimbingStairs/climbStairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n ==  0) return 0;
        int s1 = 1, s2 = 1;
        int ans = s1;
        for (int i = 2; i <= n; i++) {
            ans = s1 + s2;
            s1 = s2;
            s2 = ans;
        }
        return ans;
    }
};

File: leetcode/ClimbingStairs/climbStairs.java
public class Solution {
    public int climbStairs(int n) {
        // Start typing your Java solution below
        // DO NOT write main() function
        int sol[] = new int[n + 1];
        sol[0] = 1;
        sol[1] = 1;
        for (int i = 2; i <= n; i++) {
            sol[i] = sol[i - 1] + sol[i - 2];
        }
        return sol[n];
    }
}
